File: Silver V/10814.py
#10814  나이순 정렬

N=int(input())

age=[]
name=[]

for i in range(N):
    a,b = input().split()
    age.append(int(a))
    name.append(b)


for i in range(N):
    for j in range(N-1):
        if age[j] > age[j+1]:
            age[j],age[j+1]=age[j+1],age[j]
            name[j],name[j+1]=name[j+1],name[j]

for i in range(N):
    print(f"{age[i]} {name[i]}")

File: Silver V/10814.cpp
//10814  나이순 정렬

#include <stdio.h>
#include <string>
#include <algorithm>

using namespace std;

struct person
{
    int age;
    int idx;
    string name;
};

person people[100000];
int N;
char temp[101];

int main() {
    scanf("%d",&N);
    for(int i=0;i<N;i++) {
        scanf("%d %s",&people[i].age,temp);
        people[i].name = temp;
        people[i].idx = i;
    }

    sort(people, people+N, [](person a, person b) {
        if(a.age==b.age) return a.idx<b.idx;
        return a.age<b.age;
    });

    //출력
    for(int i=0;i<N;i++) {
        printf("%d %s\n",people[i].age,people[i].name.c_str());
    }
    return 0;
}